src/libraries/System.Private.Xml/src/System/Xml/Schema/Inference/Infer.cs

```diff
@@ -507,7 +507,7 @@ private XmlSchemaElement AddElement(string localName, string prefix, string chil
             XmlSchemaElement returnedElement = xse; //this value will change to elementReference if childURI!= parentURI
             XmlSchema xs = null;
             bool bCreatingNewType = true;
-            if (childURI == string.Empty)
+            if (childURI.Length == 0)
```

Comment:
What guarantees childURI here is non-null?

Reply:
```suggestion
            if (childURI == string.Empty)
```